docs-src/tls.md

```diff
@@ -0,0 +1,138 @@
+   specified by routes. These backend resources will usually be Services.
+  
+With Service APIs, TLS configuration of downstream and
+upstream connections is managed independently.
+
+Please note that in case of `Passthrough` TLS mode, no TLS settings take
+effect as the TLS session from the client is NOT terminated at the Gateway.
+The rest of the document assumes that TLS is being terminated at the Gateway.
+
+## Downstream TLS
+
+Downstream TLS settings are configured using Listeners at the Gateway level.
+
+### Listener and TLS 
+
+Listeners expose the TLS setting on a per domain or sub-domain basis.
+TLS settings of a listener are applied to all domains that satisfy the
+`hostname.match` criteria.
+
+In the following example, a Gateway is configured to serve a TLS certificate
+for any request that is served a Gateway.
+
+```yaml
+listeners:
+- protocol: HTTPS # Other possible value is `TLS`
+  port: 443
+  hostname:
+    match: Any # Other possible values are `Domain` and `Exact`
+  tls:
+    mode: Terminate
+    certificateRef:
+      kind: Secret
+      group: core
+      name: default-cert
+    routeOverride:
+      certificate: Deny
+```
+
+If `hostname.match` is set to `Exact`, then the TLS settings apply to only the
+specific hostname that is set in `hostname.name`. This is the recommended way
+of configuring TLS settings in Service APIs as it does not involve any
+redirection and should suffice in the majority of use-cases.
+Please take a look at the examples below for various possibilities.
+
+### Route and TLS
+
+If `listener.tls.routeOverride.certificate` is set to `Allow`, TLS certificates
+can be configured at the Route-level. This feature is primarily meant for a
+cluster with a self-service model where Service owners bring their own TLS
+certificates. This feature also mirrors the behavior of TLS as defined in the
+Ingress v1 resource. One should use this feature only when the Gateway Admin
+wishes to delegate TLS to Service or App owners.
+
+When using this feature, please note that the TLS certificate to serve is chosen
+before an HTTPRoute is selected. This is because the TLS handshake is completed
+before an HTTP request is sent from the client.
+
+[TLS Certificate in Route](#tls-certificate-in-route) provides an example
+on how this feature can be used.
+
+### Examples
+
+#### TLS in listener
+
+In this example, Gateway is configured to serve `foo.example.com` and
+`bar.example.com` domains. The certificate for these domains are specified
+in Gateway.
+
+```
+{% include 'tls-basic.yaml' %}
+```
+
+#### Wildcard TLS listeners
+
+In this example, the Gateway is configured with wildcard certificate for
+`*.example.com` and a different certificate for `foo.example.com`.
+Since a specific match takes priority, the Gateway will serve the
+`foo-example-com-cert` for requests to `foo.example.com` and the
+`wildcard-example-com-cert` for all other requests.
```

Comment:
```suggestion
In this example, the Gateway is configured with a wildcard certificate for
`*.example.com` and a different certificate for `foo.example.com`.
Since a specific match takes priority, the Gateway will serve
`foo-example-com-cert` for requests to `foo.example.com` and
`wildcard-example-com-cert` for all other requests.
```
or
```suggestion
In this example, the Gateway is configured with a wildcard certificate for
`*.example.com` and a different certificate for `foo.example.com`.
Since a specific match takes priority, the Gateway will serve
the certificate in `foo-example-com-cert` for requests to `foo.example.com` and
the certificate in `wildcard-example-com-cert` for all other requests.
```